e2e/e2eutil/utils.go

```diff
@@ -211,6 +211,11 @@ func WaitForAllocNotPending(t *testing.T, nomadClient *api.Client, allocID strin
 func WaitForJobStopped(t *testing.T, nomadClient *api.Client, job string) {
 	_, _, err := nomadClient.Jobs().Deregister(job, true, nil)
 	require.NoError(t, err, "error deregistering job %q", job)
+
+	// sleep for 3 seconds to make sure things any related events (like Consul
+	// service de-registration) has enough time to happen, because there are
+	// tests that assert such things after stopping jobs.
+	time.Sleep(3 * time.Second)
```

Comment:
hard-coded sleep does seem pretty gross, and potential for future flakiness...

would running `DeregisterOpts()` instead with [`NoShutdownDelay`](https://github.com/hashicorp/nomad/blob/v1.7.3/api/jobs.go#L376) help here at all?  also it returns an `EvalID` -- any value from watching that?

Reply:
> hard-coded sleep does seem pretty gross, and potential for future flakiness...

I concur, although we have plenty of hard-coded sleeps all across our e2eutil package...

> would running DeregisterOpts() instead with [NoShutdownDelay](https://github.com/hashicorp/nomad/blob/v1.7.3/api/jobs.go#L376) help here at all?

I don't _think_ so. It is my understanding that `NoShutdownDelay` will simply result in `Deregister` call taking less time, but the call blocks until the deregister message is applied via Raft. 

> also it returns an EvalID -- any value from watching that?

that's an idea maybe, I'll look into it.

Reply:
7e70531